internal/caller/client.go

```diff
@@ -0,0 +1,68 @@
+package caller
+
+import (
+	"github.com/the-maldridge/locksmith/pkg/telephone"
+)
+
+const (
+	clientName = "Caller"
+)
+
+// Factory method for the machine configuration
+var (
+	NativeConfig WGConfig
+	Configurator Configurator
+)
+
+// Client represents the application client.
+type Client struct {
+	status        LocksmithStatus
+	telephone     telephone.Telephone
+	configuration WGConfig
+	company       string
+}
+
+// New returns a new WireGuard Client.
+func New() Client {
+	newClient := Client{
+		status:  Preapproved,
+		company: "",
+	}
+	return newClient
+}
+
+// GetConfig returns the configuration for the Client.
+func (c *Client) GetConfig() WGConfig {
+	return c.configuration
+}
+
+// SetTelephone sets the Telephone for the Client.
+func (c *Client) SetTelephone(newPhone telephone.Telephone) {
+	c.telephone = newPhone
+}
+
+// GetTelephone returns the company name of the Client.
+func (c *Client) GetTelephone() telephone.Telephone {
+	return c.telephone
+}
+
+// SetCompany sets the company name for the Client.
+func (c *Client) SetCompany(newCompany string) {
+	c.company = newCompany
+}
+
+// GetCompany returns the company name of the Client.
+func (c *Client) GetCompany() string {
+	return c.company
+}
+
+// GetStatus returns the locksmith status of the WireGuard client.
+func (c *Client) GetStatus() LocksmithStatus {
```

Comment:
Lets poll the server for this information, so it should wait until the telephone has been fleshed out.